File: 백준/Silver/10845. 큐/큐.cc
#include <iostream>
#include <queue>
using namespace std;
int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

   queue<int> Q;
   int n;
   cin >> n;

   while (n--) {
       string str;
       cin >> str;

       if (str == "push") {
           int input;
           cin >> input;
           Q.push(input);
       } else if (str == "front") {
           if (!Q.empty()) cout << Q.front() << "\n";
           else cout << -1 << "\n";
       } else if (str == "back") {
           if (!Q.empty()) cout << Q.back() << "\n";
           else cout << -1 << "\n";
       } else if (str == "size") {
           cout << Q.size() << "\n";
       } else if (str == "empty") {
           cout << (int )Q.empty() << "\n";
       } else if (str == "pop") {
           if (!Q.empty()) {
               cout << Q.front() << "\n";
               Q.pop();
           } else cout << -1 << "\n";
       }
   }
}

File: 백준/Silver/10845. 큐/큐.kt
class CustomQueue<T> {
    var MAX = 10000
    var head: Int = 0
    var tail: Int = 0
    var Q = IntArray(MAX)

    fun push(element: Int) {
//        if (isFull()) {
//            throw Exception()
//        }
        Q[tail++] = element
    }

    fun pop(): Int {
        if (isEmpty()) {
            return -1
        }
        return Q[head++]
    }

    fun front(): Int {
        if (isEmpty()) {
            return -1
        }
        return Q[head]
    }

    fun back(): Int {
        if (isEmpty()) {
            return -1
        }
        return Q[tail - 1]
    }

    fun size(): Int {
        return tail - head
    }

    fun isEmpty(): Boolean {
        return head == tail
    }

    fun isFull(): Boolean {
        return tail == MAX - 1
    }
}

fun main() {
    var n = readln().toInt()
    var Q = CustomQueue<Int>()
    for (i in 1..n) {
        var str = readlnOrNull()!!.split(" ")
        when (str[0]) {
            "push" -> Q.push(str[1].toInt())
            "pop" -> println(Q.pop())
            "size" -> println(Q.size())
            "empty" -> println(if (Q.isEmpty()) 1 else 0)
            "front" -> println(Q.front())
            "back" -> println(Q.back())
        }
    }
}